5.0/en/0x51-V51-OAuth2.md

```diff
@@ -14,15 +14,17 @@ There are various different personas in the OAuth process, described in more det
 | **51.1.4** | [ADDED] Verify that refresh tokens are sender-constrained or use refresh token rotation to prevent token replay attacks. Refresh token rotation prevents usage in the event of a compromised refresh token. Sender-constrained refresh tokens cryptographically binds the refresh token to a particular Client. | ✓ | ✓ | ✓ |
 | **51.1.5** | [ADDED] Verify that if a Client sends a valid PKCE "code_challenge" parameter in the authorization request, the Authorization Server enforces the correct usage of "code_verifier" at the token endpoint. | ✓ | ✓ | ✓ |
 | **51.1.6** | [ADDED] Verify that the Resource Owner password credentials grant is not used or configured by the Authorization Server. This grant type insecurely exposes the credentials of the Resource Owner to the client, increasing the attack surface of the application. | ✓ | ✓ | ✓ |
+| **51.1.7** | [ADDED] Verify that the Authorization Server does not expose URLs that forward the user's browser to arbitrary URIs obtained from a query parameter ("open redirectors") which can enable exfiltration of authorization codes and access tokens. Redirect URIs must be compared using exact string-matching. | ✓ | ✓ | ✓ |
```

Comment:
"Redirect URIs must be compared using exact string-matching.": Should this not be a separate requirement ?

Reply:
Do you mean the second sentence to be separated and create another requirement than them both being in one requirement?

Reply:
@csfreak92: Yes. This looks like quite unrelated things to me.

Reply:
As I read it, the first part is "do not have an open-redirect vulnerability" on the authorization server (covered by 5.1.5):

The second part is for the string-match method, end would be covered by the outcome from https://github.com/OWASP/ASVS/issues/1965#issuecomment-2128948085

Reply:
Ah so we should just drop the open-redirect in this requirement and retain the string-match method. Gotcha, I can fix that, but let's wait for the result of that thread's comment. :)

Reply:
So this requirement got in via #1965 / #2020